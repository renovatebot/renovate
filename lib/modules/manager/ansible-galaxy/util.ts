import { regEx } from '../../../util/regex';

export const newBlockRegEx = /^\s*-\s*((\w+):\s*(.*))$/;
export const blockLineRegEx = /^\s*((\w+):\s*(.*))$/;
export const galaxyDepRegex = /[\w-]+\.[\w-]+/;
export const dependencyRegex = /^dependencies:/;
export const galaxyRegEx = regEx(
  /^\s+(?<packageName>[\w.]+):\s*["'](?<version>.+)["']\s*/
);
export const nameMatchRegex = regEx(
  /(?<source>((git\+)?(?:(git|ssh|https?):\/\/)?(.*@)?(?<hostname>[\w.-]+)(?:(:\d+)?\/|:))(?<depName>[\w./-]+)(?:\.git)?)(,(?<version>[\w.]*))?/
);
