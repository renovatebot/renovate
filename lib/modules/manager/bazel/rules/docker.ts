import { z } from 'zod';
import { DockerDatasource } from '../../../datasource/docker';
import { id as dockerVersioning } from '../../../versioning/docker';
import type { PackageDependency } from '../../types';

export const dockerRules = ['container_pull'] as const;

export const DockerTarget = z
  .object({
    rule: z.enum(dockerRules),
    name: z.string(),
    tag: z.string(),
    digest: z.string(),
    repository: z.string(),
    registry: z.string(),
  })
  .transform(
    ({ rule, name, repository, tag, digest, registry }): PackageDependency => ({
      datasource: DockerDatasource.id,
      versioning: dockerVersioning,
      depType: rule,
      depName: name,
      packageName: repository,
      currentValue: tag,
      currentDigest: digest,
      registryUrls: [registry],
    })
  );
