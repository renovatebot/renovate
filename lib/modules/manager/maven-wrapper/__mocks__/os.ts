let mockPlatform = 'darwin';
const os = jest.createMockFromModule('os');

function platform() {
    return mockPlatform
}

function cpus() {
    return [];
}

os.platform = platform;
os.cpus = cpus;
os.__setPlatform = (platform: string) => mockPlatform = platform;

module.exports = os;