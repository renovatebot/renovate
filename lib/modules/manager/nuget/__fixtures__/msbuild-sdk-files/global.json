{
  "sdk": {
    "version": "5.0.302",
    "rollForward": "latestMajor"
  },
  "msbuild-sdks": {
    "YoloDev.Sdk": "0.2.0"
  }
}
