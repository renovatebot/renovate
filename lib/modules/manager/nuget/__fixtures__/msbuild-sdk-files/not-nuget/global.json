{
  "type": "not a nuget global.json"
}
