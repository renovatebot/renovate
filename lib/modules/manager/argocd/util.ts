import { regEx } from '../../../util/regex';

export const keyValueExtractionRegex = regEx(
  /^\s*(?<key>[^\s]+):\s+"?(?<value>[^"\s]+)"?\s*$/
);
// looks for `apiVersion: argoproj.io/
export const fileTestRegex = regEx(/\s*apiVersion:\s*argoproj.io\/\s*/);
