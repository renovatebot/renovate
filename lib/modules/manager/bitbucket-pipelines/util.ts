import { regEx } from '../../../util/regex';
import { BitbucketTagsDatasource } from '../../datasource/bitbucket-tags';
import { getDep } from '../dockerfile/extract';
import type { PackageDependency } from '../types';

export const pipeRegex = regEx(`^\\s*-\\s?pipe:\\s*'?"?([^\\s'"]+)'?"?\\s*$`);
export const dockerImageRegex = regEx(
  `^\\s*-?\\s?image:\\s*'?"?([^\\s'"]+)'?"?\\s*$`
);
export const dockerImageObjectRegex = regEx('^(?<spaces>\\s*)image:\\s*$');

export function addDepAsBitbucketTag(
  deps: PackageDependency[],
  pipe: string
): void {
  const [depName, currentValue] = pipe.split(':');
  const dep: PackageDependency = {
    depName,
    currentValue,
    datasource: BitbucketTagsDatasource.id,
  };
  dep.depType = 'bitbucket-tags';
  deps.push(dep);
}

export function addDepAsDockerImage(
  deps: PackageDependency[],
  currentDockerImage: string
): void {
  const dep = getDep(currentDockerImage);
  dep.depType = 'docker';
  deps.push(dep);
}

export function addDepFromObject(
  deps: PackageDependency[],
  lines: string[],
  start: number,
  len: number,
  spaces: string
): number {
  const nameRegex = regEx(
    `^${spaces}\\s+name:\\s*['"]?(?<image>[^\\s'"]+)['"]?\\s*$`
  );
  const indentRegex = regEx(`^${spaces}\\s+`);

  for (let idx = start + 1; idx < len; idx++) {
    const line = lines[idx];

    if (!indentRegex.test(line)) {
      // malformed
      return idx;
    }

    const groups = nameRegex.exec(line)?.groups;
    if (groups) {
      const dep = getDep(groups.image);
      dep.depType = 'docker';
      deps.push(dep);
      return idx;
    }
  }

  // malformed
  return start;
}
