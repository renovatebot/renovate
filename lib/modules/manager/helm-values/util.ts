import { hasKey } from '../../../util/object';
import { regEx } from '../../../util/regex';
import type { HelmDockerImageDependency } from './types';

const parentKeyRe = regEx(/image$/i);

/**
 * Type guard to determine whether a given partial Helm values.yaml object potentially
 * defines a Helm Docker dependency.
 *
 * There is no exact standard of how Docker dependencies are defined in Helm
 * values.yaml files (as of February 26th 2021), this function defines a
 * heuristic based on the most commonly used format in the Helm charts:
 *
 * image:
 *   repository: 'something'
 *   tag: v1.0.0
 * image:
 *   repository: 'something'
 *   version: v1.0.0
 * renovateImage:
 *   repository: 'something'
 *   tag: v1.0.0
 */
export function matchesHelmValuesDockerHeuristic(
  parentKey: string,
  data: unknown
): data is HelmDockerImageDependency {
  return !!(
    parentKeyRe.test(parentKey) &&
    data &&
    typeof data === 'object' &&
    hasKey('repository', data) &&
    (hasKey('tag', data) || hasKey('version', data))
  );
}

export function matchesHelmValuesInlineImage(
  parentKey: string,
  data: unknown
): data is string {
  return !!(parentKeyRe.test(parentKey) && data && typeof data === 'string');
}
