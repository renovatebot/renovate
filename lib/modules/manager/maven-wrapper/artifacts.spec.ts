import type { Stats } from 'node:fs';
import os from 'node:os';
import type { StatusResult } from 'simple-git';
import { join } from 'upath';
import { envMock, mockExecAll } from '../../../../test/exec-util';
import { env, fs, git, mockedFunction, partial } from '../../../../test/util';
import { GlobalConfig } from '../../../config/global';
import { resetPrefetchedImages } from '../../../util/exec/docker';
import { getPkgReleases } from '../../datasource';
import { updateArtifacts } from '.';

jest.mock('../../../util/fs');
jest.mock('../../../util/git');
jest.mock('../../../util/exec/env');
jest.mock('../../datasource');

process.env.CONTAINERBASE = 'true';

const osPlatformSpy = jest.spyOn(os, 'platform');

function mockMavenFileChangedInGit(fileName = 'maven-wrapper.properties') {
  git.getRepoStatus.mockResolvedValueOnce(
    partial<StatusResult>({
      modified: [`maven.mvn/wrapper/${fileName}`],
    })
  );
}

describe('modules/manager/maven-wrapper/artifacts', () => {
  beforeEach(() => {
    osPlatformSpy.mockImplementation(() => 'linux');
    GlobalConfig.set({ localDir: join('/tmp/github/some/repo') });
    fs.statLocalFile.mockResolvedValue(
      partial<Stats>({
        isFile: () => true,
        mode: 0o555,
      })
    );

    resetPrefetchedImages();

    env.getChildProcessEnv.mockReturnValue({
      ...envMock.basic,
      LANG: 'en_US.UTF-8',
      LC_ALL: 'en_US',
    });
    mockedFunction(getPkgReleases).mockResolvedValueOnce({
      releases: [
        { version: '8.0.1' },
        { version: '11.0.1' },
        { version: '16.0.1' },
        { version: '17.0.0' },
      ],
    });
  });

  it('Should not update if there is no dep with maven:wrapper', async () => {
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven-wrapper',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'not-mavenwrapper' }],
      config: {},
    });
    expect(updatedDeps).toBeNull();
    expect(execSnapshots).toBeEmptyArray();
  });

  it('Docker should use java 8 if version is lower then 2.0.0', async () => {
    mockMavenFileChangedInGit();
    const execSnapshots = mockExecAll();
    GlobalConfig.set({ localDir: './', binarySource: 'docker' });
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: {
        currentValue: '2.0.0',
        newValue: '3.3.1',
        constraints: undefined,
      },
    });

    const expected = [
      {
        file: {
          contents: undefined,
          path: 'maven.mvn/wrapper/maven-wrapper.properties',
          type: 'addition',
        },
      },
    ];

    expect(execSnapshots[2].cmd).toContain('java 8.0.1');
    expect(updatedDeps).toEqual(expected);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('Should update when it is maven wrapper', async () => {
    mockMavenFileChangedInGit();
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { currentValue: '3.3.1', newValue: '3.3.1' },
    });

    const expected = [
      {
        file: {
          contents: undefined,
          path: 'maven.mvn/wrapper/maven-wrapper.properties',
          type: 'addition',
        },
      },
    ];
    expect(updatedDeps).toEqual(expected);
    expect(execSnapshots).toEqual([
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('Should not update deps when maven-wrapper.properties is not in git change', async () => {
    mockMavenFileChangedInGit('not-maven-wrapper.properties');
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { newValue: '3.3.1' },
    });
    expect(updatedDeps).toEqual([]);
    expect(execSnapshots).toEqual([
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('updates with docker', async () => {
    mockMavenFileChangedInGit();
    GlobalConfig.set({
      localDir: './',
      binarySource: 'docker',
      dockerSidecarImage: 'ghcr.io/containerbase/sidecar',
    });
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    const result = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { currentValue: '3.3.0', newValue: '3.3.1' },
    });
    expect(result).toEqual([
      {
        file: {
          contents: undefined,
          path: 'maven.mvn/wrapper/maven-wrapper.properties',
          type: 'addition',
        },
      },
    ]);
    expect(execSnapshots).toMatchObject([
      {
        cmd: 'docker pull ghcr.io/containerbase/sidecar',
        options: { encoding: 'utf-8' },
      },
      { cmd: 'docker ps --filter name=renovate_sidecar -aq' },
      {
        cmd:
          'docker run --rm --name=renovate_sidecar --label=renovate_child ' +
          '-v "./":"./" ' +
          '-e CONTAINERBASE_CACHE_DIR ' +
          '-w "../.." ' +
          'ghcr.io/containerbase/sidecar' +
          ' bash -l -c "' +
          'install-tool java 17.0.0 ' +
          '&& ' +
          './mvnw wrapper:wrapper"',
        options: {
          cwd: '../..',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('Should return null when cmd is not found', async () => {
    osPlatformSpy.mockImplementation(() => 'win32');
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    fs.statLocalFile.mockResolvedValue(null);
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { newValue: '3.3.1' },
    });
    expect(updatedDeps).toBeNull();
    expect(execSnapshots).toMatchObject([]);
    expect(git.getRepoStatus).not.toHaveBeenCalled();
  });

  it('Should throw an error when it cant execute', async () => {
    mockExecAll(new Error('temporary-error'));
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { currentValue: '3.0.0', newValue: '3.3.1' },
    });

    expect(updatedDeps).toEqual([
      {
        artifactError: {
          lockFile: 'maven',
          stderr: 'temporary-error',
        },
      },
    ]);
    expect(git.getRepoStatus).not.toHaveBeenCalled();
  });

  it('updates with binarySource install', async () => {
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    mockMavenFileChangedInGit();
    GlobalConfig.set({
      localDir: join('/tmp/github/some/repo'),
      binarySource: 'install',
    });
    const updatedDeps = await updateArtifacts({
      packageFileName: 'maven',
      newPackageFileContent: '',
      updatedDeps: [{ depName: 'maven-wrapper' }],
      config: { currentValue: '3.0.0', newValue: '3.3.1' },
    });

    expect(execSnapshots).toMatchObject([
      { cmd: 'install-tool java 17.0.0' },
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);

    expect(updatedDeps).toEqual([
      {
        file: {
          contents: undefined,
          path: 'maven.mvn/wrapper/maven-wrapper.properties',
          type: 'addition',
        },
      },
    ]);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('should run wrapper:wrapper with MVNW_REPOURL if it is a custom artifactory', async () => {
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    mockMavenFileChangedInGit();
    await updateArtifacts({
      packageFileName: 'maven-wrapper',
      newPackageFileContent: '',
      updatedDeps: [
        {
          depName: 'maven-wrapper',
          replaceString:
            'https://internal.local/maven-public/org/apache/maven/wrapper/maven-wrapper/3.0.0/maven-wrapper-3.0.0.jar',
        },
      ],
      config: { currentValue: '3.0.0', newValue: '3.3.1' },
    });

    expect(execSnapshots).toMatchObject([
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
            MVNW_REPOURL: 'https://internal.local/maven-public',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('should run not include MVNW_REPOURL when run with default maven repo url', async () => {
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    mockMavenFileChangedInGit();
    await updateArtifacts({
      packageFileName: 'maven-wrapper',
      newPackageFileContent: '',
      updatedDeps: [
        {
          depName: 'maven-wrapper',
          replaceString:
            'https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.1.1/maven-wrapper-3.1.1.jar',
        },
      ],
      config: { currentValue: '3.0.0', newValue: '3.3.1' },
    });

    expect(execSnapshots).toMatchObject([
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);
    expect(execSnapshots[0]!.options!.env).not.toHaveProperty('MVNW_REPOURL');
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });

  it('should run not include MVNW_REPOURL when run with a malformed replaceString', async () => {
    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
    mockMavenFileChangedInGit();
    await updateArtifacts({
      packageFileName: 'maven-wrapper',
      newPackageFileContent: '',
      updatedDeps: [
        {
          depName: 'maven-wrapper',
          replaceString: 'not a good url',
        },
      ],
      config: { currentValue: '3.0.0', newValue: '3.3.1' },
    });

    expect(execSnapshots).toMatchObject([
      {
        cmd: './mvnw wrapper:wrapper',
        options: {
          cwd: '/tmp/github',
          encoding: 'utf-8',
          env: {
            HOME: '/home/user',
            HTTPS_PROXY: 'https://example.com',
            HTTP_PROXY: 'http://example.com',
            LANG: 'en_US.UTF-8',
            LC_ALL: 'en_US',
            NO_PROXY: 'localhost',
            PATH: '/tmp/path',
          },
          maxBuffer: 10485760,
          timeout: 900000,
        },
      },
    ]);

    expect(execSnapshots[0]!.options!.env).not.toHaveProperty('MVNW_REPOURL');
    expect(git.getRepoStatus).toHaveBeenCalledOnce();
  });
});
