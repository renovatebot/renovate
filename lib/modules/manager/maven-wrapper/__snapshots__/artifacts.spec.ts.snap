// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/manager/maven-wrapper/artifacts Should not update deps when maven-wrapper.properties is not in git change 1`] = `
Array [
  Object {
    "cmd": "./mvnw wrapper:wrapper",
    "options": Object {
      "cwd": "./",
      "encoding": "utf-8",
      "env": Object {
        "HOME": "/home/user",
        "HTTPS_PROXY": "https://example.com",
        "HTTP_PROXY": "http://example.com",
        "LANG": "en_US.UTF-8",
        "LC_ALL": "en_US",
        "NO_PROXY": "localhost",
        "PATH": "/tmp/path",
      },
      "maxBuffer": 10485760,
      "timeout": 900000,
    },
  },
]
`;

exports[`modules/manager/maven-wrapper/artifacts Should return an error when config is not set 1`] = `Array []`;

exports[`modules/manager/maven-wrapper/artifacts Should return java 7.0 when current maven version is 3.0.0 1`] = `
Array [
  Object {
    "cmd": "./mvnw wrapper:wrapper",
    "options": Object {
      "cwd": "./",
      "encoding": "utf-8",
      "env": Object {
        "HOME": "/home/user",
        "HTTPS_PROXY": "https://example.com",
        "HTTP_PROXY": "http://example.com",
        "LANG": "en_US.UTF-8",
        "LC_ALL": "en_US",
        "NO_PROXY": "localhost",
        "PATH": "/tmp/path",
      },
      "maxBuffer": 10485760,
      "timeout": 900000,
    },
  },
]
`;

exports[`modules/manager/maven-wrapper/artifacts Should return java 8.0 when current maven version is lower then 3.0.0 1`] = `
Array [
  Object {
    "cmd": "./mvnw wrapper:wrapper",
    "options": Object {
      "cwd": "./",
      "encoding": "utf-8",
      "env": Object {
        "HOME": "/home/user",
        "HTTPS_PROXY": "https://example.com",
        "HTTP_PROXY": "http://example.com",
        "LANG": "en_US.UTF-8",
        "LC_ALL": "en_US",
        "NO_PROXY": "localhost",
        "PATH": "/tmp/path",
      },
      "maxBuffer": 10485760,
      "timeout": 900000,
    },
  },
]
`;

exports[`modules/manager/maven-wrapper/artifacts Should return null when cmd is not found 1`] = `Array []`;

exports[`modules/manager/maven-wrapper/artifacts Should throw an error when it cant update 1`] = `Array []`;

exports[`modules/manager/maven-wrapper/artifacts Should update when it is maven wrapper 1`] = `
Array [
  Object {
    "cmd": "./mvnw wrapper:wrapper",
    "options": Object {
      "cwd": "./",
      "encoding": "utf-8",
      "env": Object {
        "HOME": "/home/user",
        "HTTPS_PROXY": "https://example.com",
        "HTTP_PROXY": "http://example.com",
        "LANG": "en_US.UTF-8",
        "LC_ALL": "en_US",
        "NO_PROXY": "localhost",
        "PATH": "/tmp/path",
      },
      "maxBuffer": 10485760,
      "timeout": 900000,
    },
  },
]
`;
